packages/idos-sdk-js/src/lib/auth.ts

```diff
@@ -6,55 +6,60 @@ import * as Utf8Codec from "@stablelib/utf8";
 import * as BorshCodec from "borsh";
 import type { Signer } from "ethers";
 
-import { idOS } from "./idos";
+import type { Store } from "../../../idos-store";
+import type { KwilWrapper } from "./kwil-wrapper";
 import { Nonce } from "./nonce";
 import { implicitAddressFromPublicKey } from "./utils";
 
-/* global Buffer */
-
 export interface AuthUser {
   humanId: string | null;
   address: string;
   publicKey?: string;
 }
 
 export class Auth {
-  idOS: idOS;
-  user?: AuthUser;
+  private user?: AuthUser;
+
+  get currentUser() {
+    if (!this.user) throw new Error("Call idOS.setSigner first.");
+    return this.user;
+  }
+
+  constructor(public readonly kwilWrapper: KwilWrapper, public readonly store: Store) {}
 
-  constructor(idOS: idOS) {
-    this.idOS = idOS;
+  static async init(kwilWrapper: KwilWrapper, store: Store) {
+    return new Auth(kwilWrapper, store);
```

Comment:
Since we're not adding any additional logic here, I think I'd rather not have this method for now.